src/NuGet.Core/NuGet.Configuration/Utility/IReadOnlyCollectionUtility.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace NuGet.Configuration
+{
+    internal static class IReadOnlyCollectionUtility
+    {
+        internal static IReadOnlyCollection<T> Create<T>(IEqualityComparer<T> comparer, params T[] t)
+        {
+#if !NET45
+            return new HashSet<T>(t, comparer);
+#else
+            //Since we are not using a HashSet, we do not use the comparer
+            return new ReadOnlyCollection<T>(t);
```

Comment:
This is a correctness bug.

I get that [`HashSet<T>` didn't implement `IReadOnlyCollection<T>` until .NET 4.6](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.hashset-1?view=netframework-4.6#remarks), but you don't require that to ensure consistent results before and after .NET 4.5.

Consider the following:
```C#
var values = new string[] { "a", "A" };

new HashSet<string>(values, StringComparer.OrdinalIgnoreCase);  // has 1 element
new ReadOnlyCollection<string>(values);                         // has 2 elements
new ReadOnlyCollection<string>(a.ToList());                     // has 1 element
```
I think the implementation should be: 
```C#
    var hashSet = new HashSet<T>(t, comparer);

#if NET45
    return new ReadOnlyCollection<T>(hashSet.ToList());
#else
    return hashSet;
#endif
```
On .NET 4.5 there will be a tiny perf hit, but the output will be correct.

Reply:
good catch. shouldn't have let my worries be fixed by a comment.
fixed in next iteration.